### PR TITLE
Update workflows to use actions that don't need organization secrets

### PR DESCRIPTION
**Pull request by jdbaldry**
**What this PR does**:

Each repository can only have 100 organization secrets and there are now more than 100 in our organization which causes inconsitent behavior.

Some repositories don't have the secrets they need assigned.

These composite actions use secrets stored in Vault that are available to all repositories.

- `publish-technical-documentation-next.yml` has been tested with https://github.com/grafana/writers-toolkit/blob/main/.github/workflows/publish-technical-documentation.yml.
- `publish-technical-documentation-release.yml` has been tested with https://github.com/grafana/backend-enterprise/blob/gem-release-2.13/.github/workflows/publish-technical-documentation-release.yml.

There is some copy-paste involved in the creation of these workflows. Please check:

For `publish-technical-documentation-next.yml`:

- [x] The `on.push` `branches` and `paths` filters are correct for your repository.
- [x] The `jobs.sync.if` repository matches your repository.
- [x] The `jobs.sync.steps[1].with.website_directory` matches the directory you publish to in the website repository.

For `publish-technical-documentation-release.yml`:

- [x] The `on.push` `branches`, `tags`, and `paths` filters are correct for your repository.
- [x] The `jobs.sync.if` repository matches your repository.
- [x] The `jobs.sync.steps[1].with.release_tag_regexp` regular expression matches your tags and captures major, minor, and patch versions from those tags.
- [x] The `jobs.sync.steps[1].with.release_branch_regexp` regular expression matches your release branch names and captures major and minor versions from those branch names.
- [x] The `jobs.sync.steps[1].with.release_branch_with_patch_regexp` regular expression matches your release branch names if they were to include a patch version, and that it would capture major, minor, and patch versions from those branch names.
- [x] The `jobs.sync.steps[1].with.website_directory` matches the directory you publish to in the website repository.

We'll also need to backport this to any branches where you are maintaining documentation that you want synced to the website.

**Which issue(s) this PR fixes**:
Fixes [#<issue number>](https://github.com/grafana/technical-documentation/issues/970)

### CODE REVIEW

File: .github/workflows/publish-technical-documentation-next.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    if: github.repository == 'grafana/tempo'
+    permissions:
+      contents: read
+      id-token: write
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: grafana/writers-toolkit/publish-technical-documentation@publish-technical-documentation/v1
+        with:
+          website_directory: content/docs/tempo
```
Comment (jdbaldry) [inline .github/workflows/publish-technical-documentation-next.yml:21]:
```suggestion
          website_directory: content/docs/tempo/next
```

File: .github/workflows/publish-technical-documentation-release.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    if: github.repository == 'grafana/tempo'
+    permissions:
+      contents: read
+      id-token: write
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - uses: grafana/writers-toolkit/publish-technical-documentation-release@publish-technical-documentation-release/v1
```
Comment (mdisibio) [inline .github/workflows/publish-technical-documentation-release.yml:23]: Just for my understanding, it looks like there are separate actions for next and versioned updates.

`publish-technical-documentation` takes the destination folder entirely `content/docs/tempo/next`
`publish-technical-documentation-release` takes the parent folder `content/docs/tempo` and adds the version internally.

**Review by mdisibio — APPROVED** [review]
Went through the checklist (thank you!) and all looks good.  Didn't review the upstream actions but can if we'd like.

File: .github/workflows/publish-technical-documentation-release.yml (hunk 2 above)
Comment (jdbaldry) [reply .github/workflows/publish-technical-documentation-release.yml:23]: You are exactly right both about the two separate actions and the way they behave. I may consolidate them later but I can do that internally.

**Pull request merged**